File: csharp/etl/Etl.cs
using System;
using System.Linq;
using System.Collections.Generic;

public static class Etl
{
    public static Dictionary<string, int> Transform(Dictionary<int, string[]> old)
        => old
            .SelectMany(o => o.Value.Select(character => (key: character.ToLower(), value: o.Key)))
            .Distinct().ToDictionary(o => o.key, o => o.value);
}

File: go/etl/etl.go
package etl

import "strings"

// Transform changes the old scoring map format into the new format
func Transform(input map[int][]string) map[string]int {
	result := make(map[string]int)
	for key, chars := range input {
		for _, char := range chars {
			result[strings.ToLower(char)] = key
		}
	}
	return result
}
